### PR TITLE
Adds signals support

### PR DESCRIPTION
**Pull request by emersion**

### CODE REVIEW

**Comment by esiqveland** [comment]
Hey, thanks, this is really cool! Will have a look :)

File: notification.go [hunk 1]
```diff
@@ -171,6 +176,56 @@ func (n *notifier) GetServerInformation() (ServerInformation, error) {
...
+type NotificationClosedSignal struct {
+	Id uint32
+	Reason uint32
+}
+
+func (n *notifier) NotificationClosed(closed chan<- *NotificationClosedSignal) {
+	c := make(chan *dbus.Signal, len(closed))
+
+	go (func() {
+		for {
```
Comment (esiqveland) [inline notification.go:188]: Hmm, I think there needs to be a way for this go routine to exit, or it will leak?

Comment (emersion) [inline notification.go:188]: Yes, I could make the function return a channel. When the channel is closed, call `n.conn.Signal(c)` another time to remove it and stop the routine. What do you think?

Comment (esiqveland) [inline notification.go:188]: That sounds good :)

**Comment by esiqveland** [comment]
I got an idea I wanted to try out, I'm thinking it could look sort of like this: https://github.com/esiqveland/notify/pull/2

Thanks for some great ideas!

What do you think?

**Comment by emersion** [comment]
Your implementation is better :-)

It could be cool to be able to get the `NotificationClosed` and `ActionInvoked` methods for a specific notification ID too.

**Pull request closed without merging**